src/components/screens/ScheduleScreen.jsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import SuppliersToBeUpdated from "../SuppliersToBeUpdated";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  getMatchedSuppliersInContentful,
+  getSuppliersNotInContentful,
+} from "../../selectors";
+import { useEffect } from "react";
+import {
+  PROCESSED_SUPPLIERS,
+  PROCESSING_SUPPLIERS,
+} from "../../constants/app-status";
+import { updateSupplier, createSupplier } from "../../ContentfulWrapper";
+import { setSupplier } from "../../state/supplierSlice";
+import {
+  CONTENTFUL_PUT_ERROR,
+  TO_BE_PUBLISHED,
+} from "../../constants/supplier-status";
+import { setAppStatus } from "../../state/appStatusSlice";
+import { Box, Heading, Paragraph, Table } from "@contentful/f36-components";
+import SuppliersToBeCreated from "../SuppliersToBeCreated";
+import SuppliersToBeUnpublished from "../SuppliersToBeUnpublished";
+
+const ScheduleScreen = () => {
+  const dispatch = useDispatch();
+  const status = useSelector((state) => state.appStatus.value);
+
+  const suppliersToBeUpdated = useSelector(getMatchedSuppliersInContentful);
+  const suppliersToBeCreated = useSelector(getSuppliersNotInContentful);
+
+  useEffect(() => {
+    if (status === PROCESSING_SUPPLIERS) {
+      suppliersToBeUpdated.forEach(async (pair) => {
```

Comment:
this looks like a mixture of async styles within JS, as this function is declared `async` but nothing is `await`ed inside of it (instead promise style is used) - similarly nothing `await`s this function so it's a bit superfluous. If this is fire and forget then we can get rid of the `async` declaration, but if it's important that this function completes before the next one does (I notice that the despatch is done without guaranteeing these promises have resolved) then rather than promises, then `updateSupplier` should be awaited and wrapped with a try-catch, and this loop changed to `await Promise.all(suppliersToBeUpdated.map(async (pair) => {`

Reply:
Ooops - I've deleted the `async`.  It's correct that the `dispatch` calls are made as the update for each supplier completes, we don't need to wait for them all to be done.